distribution/src/bin/elasticsearch-env

```diff
@@ -122,11 +122,25 @@ if [[ "$ES_DISTRIBUTION_TYPE" == "docker" ]]; then
   while IFS='=' read -r envvar_key envvar_value
   do
     # Elasticsearch settings need to have at least two dot separated lowercase
-    # words, e.g. `cluster.name`
-    if [[ "$envvar_key" =~ ^[a-z0-9_]+\.[a-z0-9_]+ ]]; then
-      if [[ ! -z $envvar_value ]]; then
+    # words, e.g. `cluster.name`, or uppercased with underscore separators and
+    # prefixed with `ES_`, e.g. `ES_CLUSTER_NAME`. Underscores in setting names
+    # are escaped by writing them as a double-underscore e.g. "__"
+    if [[ ! -z "$envvar_value" ]]; then
+      if [[ "$envvar_key" =~ ^[a-z0-9_]+\.[a-z0-9_]+ ]]; then
         es_opt="-E${envvar_key}=${envvar_value}"
         es_arg_array+=("${es_opt}")
+      elif [[ "$envvar_key" =~ ^ES(_{1,2}[A-Z]+)+$ ]]; then
+        case "$envvar_key" in
+         ES_JAVA_HOME|ES_PATH_CONF|ES_HOME|ES_JAVA_OPTS|ES_LOG_STYLE|ES_SD_NOTIFY|ES_KEYSTORE_PASSPHRASE_FILE|ES_TMPDIR)
```

Comment:
I'd certainly prefer not to have to manually maintain this list here. Perhaps we could use a unique prefix that doesn't overlap with other environment variables?

Reply:
I agree in principal, but I've racked my brain and I'm struggling to come up with anything better. Any ideas?

| **Prefix** | **Expands to** | **Drawback** |
|------------|----------------|--------------|
| `ESS_` | ES setting | Sounds like the ESS service |
| `E_` | Elasticsearch | Sounds like an error name |
| `ES_SETTING_` | Self explanatory | So verbose! |
| `ES_CONFIG_` | Self explanatory | So verbose! |
| `ES_CFG_` | Self explanatory | Still verbose, though a little better |
| No prefix! | Same as what Kibana accepts | Can't avoid a collision with other programs |

I liked the consistency of `ES_` for vars that relate to ES - even if that means some maintenance overhead. I don't imagine us adding env vars very often?

For what it's worth I search for var names with this:

```sh
rg -g '!**/*.{java,grooy,gradle}' -g '!**/*.asciidoc' -g '!**/*.bat' '\bES_' | perl -ne 'print "$1\n" if /\b(ES_\w+)/' | sort -u
```

Reply:
> I don't imagine us adding env vars very often?

Probably not. And if/when we do, and we don't update the list it _should_ explode on us during testing.

I also cannot think of a good non-verbose alternative. Let's push forward with this as-is then.